tests/common/plugins/sanity_check/checks.py

```diff
@@ -0,0 +1,110 @@
+import json
+import logging
+import time
+
+from common.utilities import wait
+
+logger = logging.getLogger(__name__)
+SYSTEM_STABILIZE_MAX_TIME = 300
+
+
+def check_services(dut):
+    logger.info("Checking services status...")
+
+    networking_uptime = dut.get_networking_uptime().seconds
+    timeout = max((SYSTEM_STABILIZE_MAX_TIME - networking_uptime), 0)
+    interval = 20
+    logger.info("networking_uptime=%d seconds, timeout=%d seconds, interval=%d seconds" % \
+                (networking_uptime, timeout, interval))
+
+    check_result = {"failed": True}
+    if timeout == 0:    # Check services status, do not retry.
+        services_status = dut.critical_services_status()
+        check_result["failed"] = False if all(services_status.values()) else True
+        check_result["services_status"] = services_status
+    else:               # Retry checking service status
+        start = time.time()
+        elapsed = 0
+        while elapsed < timeout:
+            services_status = dut.critical_services_status()
+            check_result["failed"] = False if all(services_status.values()) else True
+            check_result["services_status"] = services_status
+
+            if check_result["failed"]:
+                wait(interval, msg="Not all services are started, wait %d seconds to retry. Remaining time: %d %s" % \
+                     (interval, int(timeout - elapsed), str(check_result["services_status"])))
+                elapsed = time.time() - start
+            else:
+                break
+
+    logger.info("Done checking services status.")
+    return check_result
+
+
+def _find_down_ports(dut, interfaces):
+    down_ports = []
+    intf_facts = dut.interface_facts()['ansible_facts']
+    for intf in interfaces:
+        try:
+            port = intf_facts["ansible_interface_facts"][intf]
+            if not port["link"] or not port["active"]:
+                down_ports.append(intf)
+        except KeyError:
+            down_ports.append(intf)
+    return down_ports
+
+
+def check_interfaces(dut):
+    logger.info("Checking interfaces status...")
+
+    networking_uptime = dut.get_networking_uptime().seconds
+    timeout = max((SYSTEM_STABILIZE_MAX_TIME - networking_uptime), 0)
+    interval = 20
+    logger.info("networking_uptime=%d seconds, timeout=%d seconds, interval=%d seconds" % \
+                (networking_uptime, timeout, interval))
+
+    mg_facts = dut.minigraph_facts(host=dut.hostname)['ansible_facts']
```

Comment:
Could we use config_facts instead of minigraph_facts?

Reply:
Yes, we can. But could you please elaborate why? Are we deprecating minigraph_facts?

Reply:
Short answer: due to BSL testing

If we configure switch with 'l2' mode with 
`sonic-cfggen -H -p -k  $HWSKU --preset l2`
only the config_db will contain new expected config, and the minigraph will be outdated

Reply:
OK, I’ll change it to use config_facts.

Reply:
Just pushed a commit replacing minigraph_facts with config_facts.